saleor/product/utils/variants_picker.py

```diff
@@ -37,29 +49,179 @@ def _attributes_to_map(instance: ProductVariant) -> Dict[str, str]:
     return attribute_map
 
 
+@dataclass
+class VariantValues:
+    pk: str
+    name: str
+    slug: str
+
+
+@dataclass
+class VariantAttributes:
+    pk: str
+    name: str
+    slug: str
+    values: List[VariantValues]
+
+    def as_dict(self):
+        return {
+            "pk": self.pk,
+            "name": self.name,
+            "slug": self.slug,
+            "values": [asdict(v) for v in self.values],
+        }
```

Comment:
Do we need that? It seems that `asdict` from `typing` do exactly the same.

Reply:
```
>>> @dataclass
... class Z:
...   field: int
...
>>> @dataclass
... class A:
...   a_field: Z
...
>>> asdict(A(Z(3)))
{'a_field': {'field': 3}}
```

I swear to god it does not worked like this last week. 
I'll remove it then...